python/ray/ml/preprocessors/batch_mapper.py

```diff
@@ -0,0 +1,29 @@
+from typing import Callable, TYPE_CHECKING
+
+from ray.ml.preprocessor import Preprocessor
+
+if TYPE_CHECKING:
+    import pandas
+
+
+class BatchMapper(Preprocessor):
+    """Apply ``fn`` to batches of records of given dataset.
+
+    This is meant to be generic and supports low level operation on records.
+    One could easily leverage this preprocessor to achieve operations like
+    adding a new column or modifying a column in place.
+
+    Args:
+        fn: The udf function for batch operation.
+    """
+
+    _is_fittable = False
+
+    def __init__(self, fn: Callable[["pandas.DataFrame"], "pandas.DataFrame"]):
+        self.fn = fn
+
+    def _transform_pandas(self, df: "pandas.DataFrame") -> "pandas.DataFrame":
+        return df.transform(self.fn)
+
+    def __repr__(self):
+        return f"<BatchMapper udf={self.fn}>"
```

Comment:
nit: Does `self.fn` actually turn into a readable string?

Reply:
it can look like: 
```
<BatchMapper udf=<function test_batch_mapper.<locals>.add_and_modify_udf at 0x7fb409149290>>
```

I don't think it is a big issue. What is the alternative?

Reply:
Ah didn't realize it would come out like that. Seems okay for now, though we should probably change this (or implement `__str__`) for all preprocessors to something more concise since this will get included in the Tune Trial name.

Reply:
Could do 
```suggestion
        return f"<BatchMapper udf={self.fn.__name__}>"
```
A little bit nicer.

Reply:
updated.